hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/schemas/V052AddressBookSchema.java

```diff
@@ -0,0 +1,105 @@
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.List;
+import java.util.Set;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * General schema for the addressbook service
+ * {@code V052AddressBookSchema} is used for migrating the address book on Version 0.52.0
+ */
+public class V052AddressBookSchema extends Schema {
+    private static final Logger log = LogManager.getLogger(V052AddressBookSchema.class);
+
+    private static final long MAX_NODES = 100L;
+    private static final SemanticVersion VERSION =
+            SemanticVersion.newBuilder().major(0).minor(52).patch(0).build();
+
+    public V052AddressBookSchema() {
+        super(VERSION);
+    }
+
+    @NonNull
+    @Override
+    public Set<StateDefinition> statesToCreate() {
+        return Set.of(StateDefinition.onDisk(NODES_KEY, EntityNumber.PROTOBUF, Node.PROTOBUF, MAX_NODES));
+    }
+
+    @Override
+    public void migrate(@NonNull final MigrationContext ctx) {
+        requireNonNull(ctx);
+        final WritableKVState<EntityNumber, Node> writableNodes =
+                ctx.newStates().get(NODES_KEY);
+        final var networkInfo = ctx.networkInfo();
+        final var nodeConfig = ctx.configuration().getConfigData(NodesConfig.class);
+        final var addressBook = networkInfo.addressBook();
+        long migratedCount = 0;
+        log.info("Started migrating nodes from address book");
+
+        for (var nodeInfo : addressBook) {
+            while (migratedCount < nodeConfig.maxNumber()) {
+                final var node = Node.newBuilder()
+                        .nodeId(nodeInfo.nodeId())
+                        .accountId(nodeInfo.accountId())
+                        .description(nodeInfo.memo())
+                        .gossipEndpoint(List.of(
+                                ServiceEndpoint.newBuilder()
+                                        .ipAddressV4(Bytes.wrap(nodeInfo.internalHostName()))
+                                        .port(nodeInfo.internalPort())
+                                        .build(),
+                                ServiceEndpoint.newBuilder()
+                                        .ipAddressV4(Bytes.wrap(nodeInfo.externalHostName()))
+                                        .port(nodeInfo.externalPort())
+                                        .build()))
+                        .gossipCaCertificate(nodeInfo.sigCertBytes())
+                        .weight(nodeInfo.stake())
+                        .build();
+                writableNodes.put(
+                        EntityNumber.newBuilder().number(nodeInfo.nodeId()).build(), node);
+                migratedCount++;
+            }
+        }
+        if (migratedCount < addressBook.size()) {
+            log.warn(
+                    "Address book contains more nodes {} than the migrated count {}",
+                    addressBook.size(),
+                    migratedCount);
```

Comment:
why would this even happen ? `migratedCount` is incremented and going  through `addressBook`

Reply:
If nodeConfig.maxNumber()<addressBook.size(), this will happen. com.hedera.node.app.service.addressbook.impl.test.schemas.V052AddressBookSchemaTest#migrateLogWarn tested this case.

Reply:
Removed nodeConfig.maxNumber() check.